targeting/geoip2/geoip2.go

```diff
@@ -123,6 +140,50 @@ func (g *GeoIP2) get(t geoType, db string) (*geoip2.Reader, error) {
 	return g.open(t, db)
 }
 
+func (g *GeoIP2) watchFiles() {
+	// Not worried about goroutines leaking because only one geoip2.New call is made in geodns.go (outside of testing)
+	ticker := time.NewTicker(1 * time.Minute)
+	go func() {
+		for {
+			select {
+			case <-ticker.C:
+				g.checkForUpdate()
+			default:
```

Comment:
without this wouldn't it just block? What's the logging mentioned?

Reply:
no CPU should be spent waiting on the channel; the for{} loop only loops fast when the `default:` case is there.

Example: https://play.golang.org/p/3TMh5Ak_WdK -- it only loops the for loop when there's a tick. If you change it to sleep for much longer you'll see no CPU is spent while waiting.

Reply:
Updated in 1942a4d330c1cd7fb3cdda6aa1b819f8c41c9c33